docs/api_reference/src/quantities.tex

```diff
@@ -30,72 +34,600 @@
 \rSec1[mp.units.core.syn]{Module \tcode{mp_units.core} synopsis}
 \indexmodule{mp_units.core}%
 \begin{codeblock}
+// mostly freestanding
 export module mp_units.core;
 
 import std;
 
 export namespace mp_units {
 
-export enum class quantity_character { scalar, vector, tensor };
+// \ref{qty.utils}, utilities
 
-// \ref{qty.traits}, traits
+// \ref{qty.sym.txt}, class template \tcode{symbol_text}
 
-template<typename Rep>
-constexpr bool treat_as_floating_point = std::is_floating_point_v<Rep>;
+enum class @\libglobal{character_set}@ : std::int8_t { utf8, portable, default_character_set = utf8 };
 
-template<typename Rep>
-constexpr bool is_scalar =
-  std::is_floating_point_v<Rep> || (std::is_integral_v<Rep> && !is_same_v<Rep, bool>);
+template<std::size_t N, std::size_t M>
+class symbol_text;
 
-template<typename Rep>
-constexpr bool is_vector = false;
+// \ref{qty.sym.expr}, symbolic expressions
 
-template<typename Rep>
-constexpr bool is_tensor = false;
+// \ref{qty.sym.expr.types}, types
 
-template<typename Rep>
-struct quantity_values;
+template<typename T, typename... Ts>
+struct per;
+
+template<typename F, int Num, int... Den>
+  requires @\seebelownc@
+struct power;
+
+// \ref{qty.ref}, reference
+
+// \ref{qty.dim}, dimension
+
+// \ref{qty.dim.concepts}, concepts
+
+template<typename T>
+concept Dimension = @\seebelownc@;
+
+template<typename T, auto D>
+concept DimensionOf = @\seebelownc@;
+
+// \ref{qty.dim.types}, types
+
+template<symbol_text Symbol>
+struct base_dimension;
+
+template<@\exposconceptnc{SymbolicConstant}@... Expr>
+struct derived_dimension;
+
+struct dimension_one;
+inline constexpr dimension_one @\libglobal{dimension_one}@{};
+
+// \ref{qty.dim.ops}, operations
+
+consteval @\libconcept{Dimension}@ auto inverse(@\libconcept{Dimension}@ auto d);
+
+template<std::intmax_t Num, std::intmax_t Den = 1, @\libconcept{Dimension}@ D>
+  requires(Den != 0)
+consteval @\libconcept{Dimension}@ auto pow(D d);
```

Comment:
This kind of ratio in template parameters only require that `Den != 0`.
However, `ratio{Num, Den}` would fail when `Abs` or `Den` is `INTMAX_MAX` due to `abs`.
Is that a problem in any of these interfaces?

Reply:
I don't know. I did not think about it so far. We may need dedicated Issues to track things like that.